InvoiceShelf/InvoiceShelf
# Adding Flat Tax support with fixed amount

# Add Support for Fixed Amount (Flat) Tax Types

## Overview
This PR introduces the ability to add and manage fixed amount (flat) taxes in the system, expanding our tax calculation capabilities beyond percentage-based taxes.

## Key Changes
- Added support for 'fixed' calculation type in tax types
- Enhanced tax display to show either percentage or fixed amount based on calculation type
- Updated tax amount calculations to handle both percentage and fixed amount scenarios
- Modified tax type selector to display formatted fixed amounts in the local currency

## UI Changes
- Tax type dropdown now shows:
  - Percentage taxes as: "Tax Name (10%)"
  - Fixed taxes as: "Tax Name ($10.00)"

## Technical Details
- Added `calculation_type` field support ('fixed' or 'percentage')
- Fixed amount values are stored in cents (integer) and displayed in decimal format
- Tax calculations now branch between:
  - Fixed: Uses the predefined fixed_amount
  - Percentage: Calculates based on item price * percentage

## Testing Instructions
1. Create a new tax type with fixed amount
2. Add the fixed tax to an item
3. Verify the tax amount is correctly applied regardless of item price
4. Test both fixed and percentage taxes on the same item
5. Verify tax calculations in invoices/estimates

## Screenshot
![Capture d’écran 2024-12-12 à 16 54 20](https://github.com/user-attachments/assets/4137f259-d2ca-4b72-b99f-9fcdb02fe11c)

## Related Issues
Closes #249
